### ISSUE TITLE
Option to skip diagnostics on vendor-like dirs 

### ISSUE THREAD

**Issue by dr0bz** [phpactor/phpactor#2688]
Hello,

there was an integration PR #2407 for this. But setting the phpstan config is ignored:

```
"language_server_phpstan.config": "/home/PROJECT_PATH/phpstan.neon",
```

phpstan.neon:
```
parameters:
  level: 8
  excludePaths:
    analyse:
     - vendor/*
```

Test file: `vendor/zircote/swagger-php/src/Annotations/Operation.php`

If i open it in vim - i get phpstan errors from phpactor. If i analyse the file from cli through phpstan - i don't get any errors, since it's excluded.

phpactor log:

`[2024-07-04 10:52:55] phpactor.DEBUG: Phpstan completed in 0.5361: '/usr/bin/php' '/home/PROJECT_PATH/vendor/bin/phpstan' 'analyse' '--no-progress' '--error-format=json' '/tmp/vMhAuSr/phpstanlsy9pvrt' '--configuration=/home/PROJECT_PATH/phpstan.neon' '--memory-limit=1G' in /home/german/workspace/rosfines-backend4_105 {"channel":"phpstan"} []`

cli:
`vendor/bin/phpstan analyse --configuration=/home/PROJECT_PATH/phpstan.neon vendor/zircote/swagger-php/src/Annotations/Operation.php
`
-> no errors.

P.S.: thanks for this great tool!!

**Comment by dantleech** [comment]
hi, thanks! are you sure they are PHPStan errors? Phpactor also produces diagnostics which can sometimes be confused with Phpstan ones. The fact that we can see the `--configuration` option being passed to PHPStan discounts the possiblity that it's not using it.

**Comment by dr0bz** [comment]
Hi,
the error comes from phpstan as far as i understand the error window correctly:

![screenshot-2024-58-09--11:20:11](https://github.com/phpactor/phpactor/assets/394521/e8f1dc48-a11b-4054-9ed9-949b79b450ee)

**Comment by dantleech** [comment]
hm. do you have any other plugins running?

phpactor is clearly running:

```
/usr/bin/php' '/home/PROJECT_PATH/vendor/bin/phpstan' 'analyse' '--no-progress' '--error-format=json' '/tmp/vMhAuSr/phpstanlsy9pvrt' '--configuration=/home/PROJECT_PATH/phpstan.neon' '--memory-limit=1G'
```

so the config is not ignored (and it's also the default location?)

**Comment by dr0bz** [comment]
I found the problem. 

The file is copied to `/tmp/vo920Om/phpstanlsKH85Mb`. The exclude of
```
  excludePaths:
    analyse:
     - vendor/*
```
will just not apply: `/tmp != vendor/*`. 

Ideas? Is a copy to /tmp necessary?

**Comment by dantleech** [comment]
yes - it's a temporary file and represents the text in your _buffer_ not the text on the disk.

maybe we could take into account the exclude paths, but not sure it's that easy.

**Comment by dr0bz** [comment]
> should we add an option to ignore vendor folder (or folders like it) for analysis

Good idea. Diagnostics in vendor like code do not bring much.

**Comment by dr0bz** [comment]
here is also an example of Phpactor diagnostics in a vendor class:
![screenshot-2024-13-10--14:31:02](https://github.com/phpactor/phpactor/assets/394521/f6e31225-1afc-4134-85f0-267a88e313d1)

**Comment by dantleech** [comment]
Should be fixed in #2705
